### PR TITLE
added support for application roles, groups

### PR DESCRIPTION
**Pull request by bvanderveen**

### CODE REVIEW

**Comment by galxy25** [comment]
@bvanderveen looks good, once the tests are added I can approve, and once the commit is signed it can be merged.

**Comment by galxy25** [comment]
also @bvanderveen if you use ` tozny/engineering` as the pull request reviewer it will automatically pull in all of us, should save you a few keystrokes and lead to faster pr turnaround time

**Review by efabens — APPROVED** [review]

**Review by galxy25 — APPROVED** [review]

**Pull request merged**
